db/db_iter.cc

```diff
@@ -709,6 +732,12 @@ bool DBIter::FindValueForCurrentKey() {
   size_t num_skipped = 0;
   while (iter_->Valid() && ikey.sequence <= sequence_ &&
          user_comparator_->Equal(ikey.user_key, saved_key_.GetKey())) {
+    if (TooManyInternalKeysSkipped()) {
+      return false;
+    } else {
+      num_internal_keys_skipped_++;
+    }
+
```

Comment:
That's a reasonable suggestion. I believe I added this code snippet here as the other check (in line 742) is also here, and I wanted to keep both of them close together.